testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -11407,3 +11407,158 @@ fn rbf_on_config_change() {
+
+    // only subscribe to the block proposal events
+    test_observer::spawn();
+    test_observer::register(&mut conf, &[EventKeyType::MinedBlocks]);
+
+    let mut btcd_controller = BitcoinCoreController::new(conf.clone());
+    btcd_controller
+        .start_bitcoind()
+        .expect("Failed starting bitcoind");
+    let mut btc_regtest_controller = BitcoinRegtestController::new(conf.clone(), None);
+    btc_regtest_controller.bootstrap_chain(201);
+
+    let mut run_loop = boot_nakamoto::BootRunLoop::new(conf.clone()).unwrap();
+    let run_loop_stopper = run_loop.get_termination_switch();
+    let counters = run_loop.counters();
+    let Counters {
+        blocks_processed,
+        naka_submitted_commits: commits_submitted,
+        ..
+    } = run_loop.counters();
+
+    let coord_channel = run_loop.coordinator_channels();
+
+    let run_loop_thread = thread::spawn(move || run_loop.start(None, 0));
+    let mut signers = TestSigners::new(vec![signer_sk]);
+    wait_for_runloop(&blocks_processed);
+    boot_to_epoch_3(
+        &conf,
+        &blocks_processed,
+        &[stacker_sk],
+        &[signer_sk],
+        &mut Some(&mut signers),
+        &mut btc_regtest_controller,
+    );
+
+    info!("------------------------- Reached Epoch 3.0 -------------------------");
+
+    blind_signer(&conf, &signers, &counters);
+
+    wait_for_first_naka_block_commit(60, &commits_submitted);
+
+    // Mine 1 nakamoto tenure
+    next_block_and_mine_commit(&mut btc_regtest_controller, 60, &conf, &counters).unwrap();
+
+    let burnchain = conf.get_burnchain();
+    let _sortdb = burnchain.open_sortition_db(true).unwrap();
+    let (_chainstate, _) = StacksChainState::open(
+        conf.is_mainnet(),
+        conf.burnchain.chain_id,
+        &conf.get_chainstate_path_str(),
+        None,
+    )
+    .unwrap();
+
+    info!("------------------------- Setup finished, run test -------------------------");
+
+    let http_origin = format!("http://{}", &conf.node.rpc_bind);
+
+    let get_v3_transactions = |txid: Txid| {
+        let url = &format!("{http_origin}/v3/transactions/{txid}");
+        info!("Send request: GET {url}");
+        reqwest::blocking::get(url)
+            .unwrap_or_else(|e| panic!("GET request failed: {e}"))
+            .json::<serde_json::Value>()
+            .unwrap()
+    };
+
+    let get_transaction_from_block = |index_block_hash: String, txid: String| {
+        for block_json in test_observer::get_blocks() {
+            if block_json["index_block_hash"].as_str().unwrap() == format!("0x{}", index_block_hash)
+            {
+                for transaction_json in block_json["transactions"].as_array().unwrap() {
+                    if transaction_json["txid"].as_str().unwrap() == format!("0x{}", txid) {
+                        return Some(transaction_json["raw_tx"].as_str().unwrap().to_string());
+                    }
+                }
+            }
+        }
+        None
+    };
```

Comment:
This is pretty nit, but I prefer the use of iterators for a situation like this and find it a bit more Rust-idiomatic

```suggestion
    let get_transaction_from_block = |index_block_hash: String, txid: String| {
        test_observer::get_blocks()
            .into_iter()
            .find(|block_json| {
                block_json["index_block_hash"].as_str().unwrap()
                    == format!("0x{}", index_block_hash)
            })
            .and_then(|block_json| {
                block_json["transactions"]
                    .as_array()
                    .unwrap()
                    .iter()
                    .find(|transaction| {
                        transaction["txid"].as_str().unwrap() == format!("0x{}", txid)
                    })
                    .map(|transaction| transaction["raw_tx"].as_str().unwrap().to_string())
            })
    };
```